restore mononoke getdeps integration tests

ahornby opened the pull request:
Bring back mononoke getdeps integration tests.  This is a refesh of the previously working export-D34186407 branch from 2022

Main changes since:
 * depend on restored sapling manifest from previous commit
 * bring back selected manifests needed by mononoke that were deleted in badf6d159dada727acfb5394bd17de223b336975. I added the sqlite binary packages from this to the sqlite3 manifest rather than bring back the sqllite3-bin manifest

Test Plan:

first build dependencies:
```
python3 ./build/fbcode_builder/getdeps.py install-system-deps --recursive mononoke_integration
python3 ./build/fbcode_builder/getdeps.py build --allow-system-packages --no-facebook-internal --src-dir=. --no-tests sapling
python3 ./build/fbcode_builder/getdeps.py build --allow-system-packages --no-facebook-internal --src-dir=. --no-tests mononoke
```

then test locally:
```
./build/fbcode_builder/getdeps.py build --num-jobs 16 --allow-system-packages --no-facebook-internal --src-dir=. mononoke_integration
./build/fbcode_builder/getdeps.py test --num-jobs 16 --allow-system-packages --no-facebook-internal --src-dir=. mononoke_integration
```

to iterate on .t:
```
./build/fbcode_builder/getdeps.py build --no-deps --num-jobs 16 --allow-system-packages --no-facebook-internal --src-dir=. mononoke_integration
./build/fbcode_builder/getdeps.py test --num-jobs 16 --allow-system-packages --no-facebook-internal --src-dir=. mononoke_integration
```

Generate the github actions yaml
```
./build/fbcode_builder/getdeps.py generate-github-actions --allow-system-packages --test --os-type linux --job-file-prefix mononoke-integration_ --job-name-prefix "Mononoke Integration " --output-dir .github/workflows mononoke_integration
```

quark-zju commented on eden/scm/exec/hgmain/src/chg.rs:
```diff
@@ -129,19 +129,30 @@ fn should_call_chg(args: &[String]) -> (bool, &'static str) {
-        }
-    } else if cfg!(unix) {
-        return (true, "unix");
+#[cfg(all(unix,not(fbcode_build)))]
```
Why this change? It seems the original code is cleaner - no duplication of the function signature.

ahornby replied:
Original version didn't compile, error due to unknown feature

quark-zju replied:
Do you have the error message? It's not clear from the code what feature is unknown (unix? fbcode_build?), or how the new code solves the problem.

ahornby replied:
can't reproduce it now.  may have been due to toolchain on my personal machine.   will try the PRs to facebook/sapling without it

quark-zju commented on eden/mononoke/tests/integration/library.sh:
```diff
@@ -2225,9 +2225,18 @@ function gitexport() {
   log="$TESTTMP/gitimport.out"
 
+  local git_cmd
+  if type --path git.real > /dev/null; then
```
Just set `NOSCMLOG=1` at some top-level? That applies to subprocesses named `git` or `sl` spawned by `git` or `sl` too.

ahornby replied:
Mononoke defaults to git.real which doesn't exist outside meta. This check detects if we are inside meta (git.real is ok) or outside (use git).

quark-zju commented:
Might be easier to review if split into smaller diffs.

ahornby committed:
restore sapling getdeps so eden/mononoke can depend on it

Summary:
Bring a manifest for sapling back so we can test eden fs and mononoke. Their tests need the hg binary and thus a sapling manifest to depend upon

How:
 * started with backout of 73302ef03f2d10e45a36aaa2354b4f722cd71e07 and badf6d159dada727acfb5394bd17de223b336975

Test Plan:

Setup a local ubuntu 22.04 toolbox with:
```
toolbox create --distro ubuntu --release 22.04
toolbox enter ubuntu-toolbox-22.04
sudo apt update
sudo apt install g++ libssl-dev pkg-config
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
```

Build locally with: `python3 build/fbcode_builder/getdeps.py --allow-system-packages build --src-dir=. sapling  --project-install-prefix sapling:/usr/local`

Test locally with: `python3 build/fbcode_builder/getdeps.py --allow-system-packages test --src-dir=. sapling  --project-install-prefix sapling:/usr/local`

Generate github actions with:

`./build/fbcode_builder/getdeps.py --allow-system-packages generate-github-actions --free-up-disk --os-type=linux --src-dir=. --output-dir=.github/workflows --job-name "Sapling CLI Getdeps " --job-file-prefix=sapling-cli-getdeps_ sapling`

CI

ahornby committed:
restore mononoke getdeps integration tests

Summary:
Bring back mononoke getdeps integration tests.  This is a refesh of the previously working export-D34186407 branch from 2022

Main changes since:
 * depend on restored sapling manifest from previous commit
 * bring back selected manifests needed by mononoke that were deleted in D51869247. I added the sqlite binary packages from this to the sqlite3 manifest rather than bring back the sqllite3-bin manifest
 * add manifests for new tools used: ripgrep, git-lfs, and zstd cli (we already had the zstd libs)
 * fix a few test expecations that where too closely tied to git cli version etc (ubuntu 22.04 is on older version)
 * fix TLS message test expecation in
 * getdeps MakefileBuilder.run_tests() improvements
   * fix error status reporting, was not failing if tests failed
    * pass of --num-jobs to tests to stop it OOMing my machine
    * pass of --filter to tests so can iterate on one test more easily

Test Plan:

first build dependencies:
```
python3 ./build/fbcode_builder/getdeps.py install-system-deps --recursive mononoke_integration
python3 ./build/fbcode_builder/getdeps.py build --allow-system-packages --no-facebook-internal --src-dir=. --no-tests sapling
python3 ./build/fbcode_builder/getdeps.py build --allow-system-packages --no-facebook-internal --src-dir=. --no-tests mononoke
```

then test locally:
```
./build/fbcode_builder/getdeps.py build --num-jobs 16 --allow-system-packages --no-facebook-internal --src-dir=. mononoke_integration
./build/fbcode_builder/getdeps.py test --num-jobs 16 --allow-system-packages --no-facebook-internal --src-dir=. mononoke_integration
```

to iterate on .t:
```
./build/fbcode_builder/getdeps.py build --no-deps --num-jobs 16 --allow-system-packages --no-facebook-internal --src-dir=. mononoke_integration
./build/fbcode_builder/getdeps.py test --num-jobs 4 --allow-system-packages --no-facebook-internal --src-dir=. mononoke_integration
```

Can test local execution for one test with:
```
python3 ./build/fbcode_builder/getdeps.py test --num-jobs 4 --allow-system-packages --no-facebook-internal --src-dir=. mononoke_integration --retry 0 --filter test-gettreepack.t
```

Generate the github actions yaml
```
./build/fbcode_builder/getdeps.py generate-github-actions --allow-system-packages --test --os-type linux --job-file-prefix mononoke-integration_ --job-name-prefix "Mononoke Integration " --output-dir .github/workflows mononoke_integration --num-jobs 4
```

ahornby commented:
closing as raised PR to https://github.com/facebook/sapling/pull/951
